crates/kernel/src/tasks/scheduler.rs

```diff
@@ -722,169 +629,172 @@ impl Scheduler {
+                    warn!(task_id, "Task not found for input request");
+                    return;
+                };
+                // Commit the session (not DB transaction) to make sure current output is
+                // flushed up to the prompt point.
+                let Ok(()) = tc.session.commit() else {
+                    warn!("Could not commit session; aborting task");
+                    return inner.send_task_result(task_id, TaskResult::Error(TaskAbortedError));
+                };
+
+                let Ok(()) = tc.session.request_input(tc.player, input_request_id) else {
+                    warn!("Could not request input from session; aborting task");
+                    return inner.send_task_result(task_id, TaskResult::Error(TaskAbortedError));
+                };
+                inner.suspended.insert(
+                    task_id,
+                    SuspendedTask {
+                        wake_condition: WakeCondition::Input(input_request_id),
+                        task,
+                        session: tc.session,
+                        result_sender: tc.result_sender,
+                    },
+                );
 
-                    let Ok(()) = task.session.request_input(task.player, input_request_id) else {
-                        warn!("Could not request input from session; aborting task");
-                        return Some(TaskHandleResult::Result(
-                            task_id,
-                            TaskResult::Error(TaskAbortedError),
-                        ));
-                    };
-                    task.waiting_input = Some(input_request_id);
-                    inner.input_requests.insert(input_request_id, task_id);
-                }
                 trace!(?task_id, "Task suspended waiting for input");
-                None
             }
-            SchedulerControlMsg::DescribeOtherTasks(reply) => {
+            SchedulerControlMsg::RequestQueuedTasks(reply) => {
                 // Task is asking for a description of all other tasks.
-                Some(TaskHandleResult::Describe(task_id, reply))
+                if let Err(e) = reply.send(self.tasks()) {
+                    error!(?e, "Could not send task description to requester");
+                    // TODO: murder this errant task
+                }
             }
             SchedulerControlMsg::KillTask {
                 victim_task_id,
                 sender_permissions,
                 result_sender,
             } => {
                 // Task is asking to kill another task.
-                Some(TaskHandleResult::Kill {
-                    requesting_task_id: task_id,
-                    victim_task_id,
-                    sender_permissions,
-                    result_sender,
-                })
+                let mut inner = self.task_q.lock().unwrap();
+                inner.kill_task(victim_task_id, sender_permissions, result_sender);
```

Comment:
Does the sender expect a reply for `KillTask`?

Reply:
Good catch, I missed that.